exports/dataset.cpp

```diff
@@ -470,9 +470,23 @@ PYBIND11_MODULE(dataset, m) {
           "data",
           &as_VariableView_variant<Variable, double, float, int64_t, int32_t,
                                    char, bool, std::string, Dataset>)
+      .def(py::self + py::self, py::call_guard<py::gil_scoped_release>())
+      .def(py::self + float(), py::call_guard<py::gil_scoped_release>())
+      .def(py::self - py::self, py::call_guard<py::gil_scoped_release>())
+      .def(py::self - float(), py::call_guard<py::gil_scoped_release>())
+      .def(py::self * py::self, py::call_guard<py::gil_scoped_release>())
+      .def(py::self * float(), py::call_guard<py::gil_scoped_release>())
+      .def(py::self / py::self, py::call_guard<py::gil_scoped_release>())
+      .def(py::self / float(), py::call_guard<py::gil_scoped_release>())
       .def(py::self += py::self, py::call_guard<py::gil_scoped_release>())
+      .def(py::self += float(), py::call_guard<py::gil_scoped_release>())
       .def(py::self -= py::self, py::call_guard<py::gil_scoped_release>())
+      .def(py::self -= float(), py::call_guard<py::gil_scoped_release>())
       .def(py::self *= py::self, py::call_guard<py::gil_scoped_release>())
+      .def(py::self *= float(), py::call_guard<py::gil_scoped_release>())
+      .def(py::self /= py::self, py::call_guard<py::gil_scoped_release>())
+      .def(py::self /= float(), py::call_guard<py::gil_scoped_release>())
```

Comment:
For maximum efficiency, we would also need overloads for `VariableSlice` as the second operand. Otherwise this uses implicit conversion, which causes a copy.